internal/addrs/parse_ref.go

```diff
@@ -396,13 +409,40 @@ func parseResourceRef(mode ResourceMode, startRange hcl.Range, traversal hcl.Tra
 	case hcl.TraverseAttr:
 		typeName = tt.Name
 	default:
-		// If it isn't a TraverseRoot then it must be a "data" reference.
-		diags = diags.Append(&hcl.Diagnostic{
-			Severity: hcl.DiagError,
-			Summary:  "Invalid reference",
-			Detail:   `The "data" object does not support this operation.`,
```

Comment:
hmm, I'm just surprised that it used multiple strategies here. I thought the names would be packed into a single constant string. Seems that some other algorithm is used when the source isn't a series like `iota`. (using `iota` would create `"InvalidResourceModeDataResourceModeEphemeralResourceModeManagedResourceMode"`)

Reply:
Yeah, it's to catch invalid traverse steps, so I think it could technically have also caught `resource.type[0]` if called directly. I think that gets validated before we get here though, so I'm not sure we could ever print the incorrect message